pkg/pipeline/encode/encode_prom.go

```diff
@@ -102,9 +104,15 @@ func (e *encodeProm) EncodeMetric(metric config.GenericMap) []config.GenericMap
 	// TODO: We may need different handling for histograms
 	out := make([]config.GenericMap, 0)
 	for metricName, mInfo := range e.metrics {
+		val, keyFound := metric[mInfo.filterKey]
+		shouldKeepRecord := keyFound && val == mInfo.filterValue
+		if !shouldKeepRecord {
+			continue
+		}
+
 		metricValue, ok := metric[mInfo.input]
 		if !ok {
-			log.Debugf("field %v is missing", metricName)
+			log.Errorf("field %v is missing", mInfo.input)
```

Comment:
I believe the reason for using Debug originally is that before introducing dedicated `filter` field, the filtering was done based on the input field. So it wasn't an error if the expected input field didn't exist. It probably indicated that the record didn't belong to the metric in context.
The input field basically had two roles:
1. hold the value we wanted to measure
2. filter records - associate a record with a metric.

because of this, prefixes were added to the input key.
This PR allows removing these prefixes and remove the 2nd role of the input field.